### PR TITLE
create multiplex.json

### PR DESCRIPTION
**Pull request by k-rister**

### CODE REVIEW

**Review by atheurer — APPROVED** [review]

File: multiplex.json [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	},
+	"positive_integer_list": {
+	    "description": "1 or more (comma separated) positive integer values",
+	    "args": [ "src-ports", "dst-ports", "vxlan-ids", "vlan-ids" ],
+	    "vals": "^[1-9][0-9]*(,[1-9][0-9]*)*$"
+	},
+	"nonezero_positive_float": {
+	    "description": "any non-zero positive floating point value",
+	    "args": [ "search-granularity", "rate", "teaching-measurement-interval", "trex-profiler-interval" ],
+	    "vals": "^([1-9][0-9]*\\.?[0-9]*)|(0?\\.[0-9]+)$"
```
Comment (rafaelfolco) [inline multiplex.json:39]: pass!
```
>>> import re
>>> re.match('^([1-9][0-9]*\\.?[0-9]*)|(0?\\.[0-9]+)$', '0.02')
<re.Match object; span=(0, 4), match='0.02'>
>>> re.match('^([1-9][0-9]*\\.?[0-9]*)|(0?\\.[0-9]+)$', '0.02')
<re.Match object; span=(0, 4), match='0.02'>
>>> re.match('^([1-9][0-9]*\\.?[0-9]*)|(0?\\.[0-9]+)$', '0.02aa')
>>> re.match('^([1-9][0-9]*\\.?[0-9]*)|(0?\\.[0-9]+)$', '-002')
>>> re.match('^([1-9][0-9]*\\.?[0-9]*)|(0?\\.[0-9]+)$', '-0.02')
>>> re.match('^([1-9][0-9]*\\.?[0-9]*)|(0?\\.[0-9]+)$', '+0.02')
>>> re.match('^([1-9][0-9]*\\.?[0-9]*)|(0?\\.[0-9]+)$', '0,02')
>>> re.match('^([1-9][0-9]*\\.?[0-9]*)|(0?\\.[0-9]+)$', '0.0.2')
>>> re.match('^([1-9][0-9]*\\.?[0-9]*)|(0?\\.[0-9]+)$', '002.1')
>>> re.match('^([1-9][0-9]*\\.?[0-9]*)|(0?\\.[0-9]+)$', '02.1')
>>> re.match('^([1-9][0-9]*\\.?[0-9]*)|(0?\\.[0-9]+)$', '2.1')
<re.Match object; span=(0, 3), match='2.1'>
```

File: multiplex.json [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+	"positive_integer": {
+	    "description": "any positive integer value",
+	    "args": [ "validation-runtime", "search-runtime", "sniff-runtime", "latency-rate",
+		      "trial-gap", "max-retries" ],
+	    "vals": "^[0-9]+$"
+	},
+	"positive_integer_list": {
+	    "description": "1 or more (comma separated) positive integer values",
+	    "args": [ "src-ports", "dst-ports", "vxlan-ids", "vlan-ids" ],
+	    "vals": "^[1-9][0-9]*(,[1-9][0-9]*)*$"
```
Comment (rafaelfolco) [inline multiplex.json:34]: pass!
```
>>> re.match("^[1-9][0-9]*(,[1-9][0-9]*)*$", '2.1')
>>> re.match("^[1-9][0-9]*(,[1-9][0-9]*)*$", '2')
<re.Match object; span=(0, 1), match='2'>
>>> re.match("^[1-9][0-9]*(,[1-9][0-9]*)*$", '02')
>>> re.match("^[1-9][0-9]*(,[1-9][0-9]*)*$", '40m')
>>> re.match("^[1-9][0-9]*(,[1-9][0-9]*)*$", '40,')
>>> re.match("^[1-9][0-9]*(,[1-9][0-9]*)*$", '40.')
```

File: multiplex.json [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    "presets": {
+	"essentials": [
+	    { "arg": "process-all-profiler-data", "vals": [ "ON" ] }
+	]
+    },
+    "validations": {
+	"pci_device_list": {
+	    "description": "1 or more (comma separated) PCI device address pairs",
+	    "args": [ "devices", "active-devices" ],
+	    "vals": "^0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9](,0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9])*$"
```
Comment (rafaelfolco) [inline multiplex.json:11]: wow 2 or more required! 
PASS!!!!
```
>>> re.match("^0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9](,0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9])*$", '17.00.1')
>>> re.match("^0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9](,0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9])*$", '17:00.1')
>>> re.match("^0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9](,0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9])*$", '0000:83:00.0')
>>> re.match("^0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9](,0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9])*$", '0000:83:00.0,0000:83:00.1')
<re.Match object; span=(0, 25), match='0000:83:00.0,0000:83:00.1'>
>>> re.match("^0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9](,0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9])*$", '0000:83:00.0,0000:83:00.1,')
>>> re.match("^0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9](,0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9])*$", '0000:83:00.,0000:83:00.1')
>>> re.match("^0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9](,0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9])*$", '0000:83:00,0000:83:00.1')
>>> re.match("^0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9](,0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9])*$", '0000:83,0000:83:00.1')
>>> re.match("^0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9](,0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9],0{4}(:[0-9a-fA-F]{2}){2}\\.[0-9])*$", '0000:83:00.1')
```

**Review by rafaelfolco — DISMISSED** [review]
looks great

**Comment by k-rister** [comment]
@rafaelfolco can you test this on your environment along with reviewing it?  you should be able to copy into in the bench-trafficgen dir and do a run and see if breaks or not.

**Comment by rafaelfolco** [comment]
It works well

**Review by rafaelfolco — APPROVED** [review]

**Pull request merged**